src/TipTapEditor/TipTapEditor.tsx

```diff
@@ -2,24 +2,39 @@ import './TipTapEditor.css';
 
 import { Editor, EditorContent } from '@tiptap/react';
 import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useDebounce } from 'usehooks-ts';
 
+import { setTextSelection } from '../features/selection/selectionSlice';
 import { EditorProps } from '../types/EditorProps';
 import { MenuBar } from './MenuBar';
 import { ReferenceNode } from './ReferenceBlock/ReferenceNode';
-import { EDITOR_EXTENSIONS, INITIAL_CONTENT } from './TipTapEditorConfigs';
+import { EDITOR_EXTENSIONS } from './TipTapEditorConfigs';
+
+export function TipTapEditor({ editorRef, editorContent }: EditorProps) {
+  const [selection, setSelection] = useState('');
 
-export function TipTapEditor({ editorRef, editorContent, onSelectionChange }: EditorProps) {
   const [editor, setEditor] = useState<Editor | null>(null);
   useEffect(() => {
-    setEditor(new Editor({
-      extensions: EDITOR_EXTENSIONS,
-      content: editorContent ?? INITIAL_CONTENT,
-      onSelectionUpdate({ editor }) {
-        const { from, to } = editor.view.state.selection;
-        onSelectionChange(editor.view.state.doc.textBetween(from, to));
-      },
-    }));
-  }, [editorContent, onSelectionChange]);
+    setEditor(
```

Comment:
We need to refactor this code (interaction with TipTap), but unfortunately their API to set/reset content isn't straightforward. CC @sehyod

Reply:
Yes, unfortunately, using `setText` leads to unwanted behaviours, especially regarding the history, because there is no easy way to clear the history (cf https://github.com/ueberdosis/tiptap/issues/491).
I think at some point we will have in-memory editors anyway, corresponding to open tabs, so the create operation would only be called once per file.